polyfills/contacts/contacts.js

```diff
@@ -0,0 +1,139 @@
+// mozContacts polyfill!
+// https://developer.mozilla.org/es/docs/Web/API/ContactManager
+
+// I have to define the service protocol, and this is as good place as any. The
+// protocol will be as follows:
+// * Each request will have a numeric request id, set by the client
+// * The answers will include the request id, so the client can know to which
+//   request the answer corresponds
+
+// For all the operations over a DeviceStorage object:
+//  * Request: { id: requestId,
+//               data: {
+//                 operation: methodName,
+//                 params: params
+//               }
+//             }
+// * Answer: { id: requestId,
+//             result|error: Whatever }
+
+// For the oncontactchange operation
+//  Request: { id: requestId,
+//             data: {
+//             operation: 'oncontactchange'
+//           },
+// Answer: Will be invoked when there's activity with
+// the MozContactChangeEvent (like) object:
+//    { id: requestId,
+//      event: MozContactChangeEvent }
+
+'use strict';
```

Comment:
Move this to line 35 (inside the function). That way you don't alter what's not in your scope